tests/client/test_base.py

```diff
@@ -34,20 +34,21 @@ def setUp(self):
         self.payload = {'foo': 'bar'}
         self.request_kwargs = {'headers': {'my': 'header'}, 'timeout': 30}
 
-    def make_retry_call_with_error_code(self, http_error_code, max_retries=None, error_json=None):
+    def make_retry_call_with_error_code(self, http_error_code, max_retries=None, error_json=None, response_headers=None):
         """
         Makes a call that will raise an http error in order to potentially trigger
-        the request being retried up to "max_retries" times.  Otherwise, a CanvasAPIError is expected
+        the request being retried up to "max_retries" times.  Otherwise, an SDK Error is expected
         to be raised by the underlying call method.  Return the exception in order to do any additional
         assertions.
         """
-        self.session.request.return_value.raise_for_status.side_effect = HTTPError(
-            response=mock.MagicMock(
-                status_code=http_error_code,
-                json=mock.Mock(return_value=error_json or {}),
-            )
-        )
-        with self.assertRaises(CanvasAPIError) as canvas_error:
+        self.session.request.return_value.raise_for_status.side_effect = HTTPError()
+        self.session.request.return_value.status_code = http_error_code
+        self.session.request.return_value.json = mock.Mock(
```

Comment:
Looks like I was just being lazy and copied over the original call with the `Mock` wrapper.  No need for that in this case though, so I'll change it.